src/sql/src/plan/query.rs

```diff
@@ -2372,9 +2467,20 @@ fn plan_table_alias(mut scope: Scope, alias: Option<&TableAlias>) -> Result<Scop
     Ok(scope)
 }
 
-fn invent_column_name(ecx: &ExprContext, expr: &Expr<Aug>) -> Option<ColumnName> {
+fn invent_column_name(
+    ecx: &ExprContext,
+    expr: &Expr<Aug>,
+    table_func_names: &HashMap<String, Ident>,
```

Comment:
Some doc comment about `table_func_names` would help here, e.g. we contrive a UUID, which we store as a string, and we need to be able to take it back to a column name.